bin/start_wifi_connect.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+IS_CONNECTED=''
+
+if [[ ! -z $CHECK_CONN_FREQ ]]
+    then
+        freq=$CHECK_CONN_FREQ
+    else
+        freq=120
+fi
+
+
+sleep 5
+
+while [[ true ]]; do
```

Comment:
This is an infinite loop. You want to make it check for if `IS_CONNECTED`'s condition and `exit 0` once it is connected.

Reply:
Doing exiting (plus using `on-failure` for the restart policy) doesn't work as expected. The container doesn't start when the Pi boots up (or at least, when the Docker daemon starts). Using `always` or `unless-stopped` might not cut it.

ATM, I can do the following:
- `break` out of the `while` loop once connected
- Execute `/usr/bin/balena-idle` at the very end of the script. (`balena-idle` is just like a script running `sleep infinity`).
  - With this, it's idle, although it's still running.

Reply:
Here's an output of using idling:

```
screenly-srly-ose-wifi-connect-1  | Checking internet connectivity ...
screenly-srly-ose-wifi-connect-1  | Your device is already connected to the internet.
screenly-srly-ose-wifi-connect-1  | Skipping setting up Wifi-Connect Access Point. Will check again in 10 seconds.
screenly-srly-ose-wifi-connect-1  | Idling...
```